src/sapling/sapling_operation.cpp

```diff
@@ -376,6 +377,7 @@ static CacheCheckResult CheckCachedNote(const SaplingNoteEntry& t, const libzcas
 
 OperationResult SaplingOperation::loadUnspentNotes(TxValues& txValues, uint256& ovk)
 {
+    AssertLockHeld(pwalletMain->cs_wallet);
```

Comment:
This is the opposite, the internal wallet methods call locks it, this shouldn't be locked for the entire execution.